src/mp-instance.js

```diff
@@ -1292,189 +1295,68 @@ function completeSDKInitialization(apiKey, config, mpInstance) {
-        config.workspaceToken
-    );
-
-    // idCache is instantiated here as opposed to when _Identity is instantiated
-    // because it depends on _Store.storageName, which is not sent until above
-    // because it is a setting on config which returns asyncronously
-    // in self hosted mode
-    mpInstance._Identity.idCache = new LocalStorageVault(
-        `${mpInstance._Store.storageName}-id-cache`,
-        {
-            logger: mpInstance.Logger,
-        }
-    );
+    mpInstance._Store.processConfig(config);
 
+    mpInstance._Identity.idCache = createIdentityCache(mpInstance);
     removeExpiredIdentityCacheDates(mpInstance._Identity.idCache);
 
-    if (config.hasOwnProperty('workspaceToken')) {
-        mpInstance._Store.SDKConfig.workspaceToken = config.workspaceToken;
+    // Web View Bridge is used for cases where Web SDK is loaded within an iOS or Android Web View
+    // on a Native Mobile Device and will not be responsible for sending events to the mParticle servers.
+    // The Web SDK will not initialize persistence or Identity directly, and simply act as a passthrough
+    // to an mParticle Native SDK.
+    if (mpInstance._Store.webviewBridgeEnabled) {
+        mpInstance._NativeSdkHelpers.initializeSessionAttributes(apiKey);
     } else {
-        mpInstance.Logger.warning(
-            'You should have a workspaceToken on your config object for security purposes.'
-        );
-    }
+        // Main SDK initialization flow
 
-    if (config.hasOwnProperty('requiredWebviewBridgeName')) {
-        mpInstance._Store.SDKConfig.requiredWebviewBridgeName =
-            config.requiredWebviewBridgeName;
-    } else if (config.hasOwnProperty('workspaceToken')) {
-        mpInstance._Store.SDKConfig.requiredWebviewBridgeName =
-            config.workspaceToken;
-    }
-    mpInstance._Store.webviewBridgeEnabled = mpInstance._NativeSdkHelpers.isWebviewEnabled(
-        mpInstance._Store.SDKConfig.requiredWebviewBridgeName,
-        mpInstance._Store.SDKConfig.minWebviewBridgeVersion
-    );
-
-    mpInstance._Store.configurationLoaded = true;
-
-    // https://go.mparticle.com/work/SQDSDKS-6044
-    if (!mpInstance._Store.webviewBridgeEnabled) {
         // Load any settings/identities/attributes from cookie or localStorage
         mpInstance._Persistence.initializeStorage();
-    }
 
-    if (mpInstance._Store.webviewBridgeEnabled) {
-        mpInstance._NativeSdkHelpers.sendToNative(
-            Constants.NativeSdkPaths.SetSessionAttribute,
-            JSON.stringify({ key: '$src_env', value: 'webview' })
+        // Set up user identitiy variables for later use
+        const currentUser = mpInstance.Identity.getCurrentUser();
+        const currentUserMPID = currentUser ? currentUser.getMPID() : null;
+        const currentUserIdentities = currentUser
+            ? currentUser.getUserIdentities().userIdentities
+            : {};
+
+        processIdentifyRequest(
```

Comment:
I think it's something we should look at from a higher level, because it still isn't clear to me why we're even keeping the identifyRequest in the store, since it probably should be part of Identity, or even its own entity entirely. I get that the initial identify request comes from the init config, which is _technically_ part of the store, but given that everything in the actual callback relies on Identity and the User object, we probably should make it part of Identity. The only thing that couples it with Store is the `isLoggedIn` bool which IMO should be part of the user.

Reply:
I'm cool w revisiting. It is currently working as is from your refactor, and your suggestion is out of scope for the current PR.